src/sentry/models/identity.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import absolute_import
+
+from django.conf import settings
+from django.db import models
+from django.utils import timezone
+
+from sentry.db.models import (
+    ArrayField, BoundedPositiveIntegerField, EncryptedJsonField,
+    FlexibleForeignKey, Model
+)
+
+
+# TODO(dcramer): pull in enum library
+class IdentityStatus(object):
+    UNKNOWN = 0
+    VALID = 1
+    INVALID = 2
+
+
+class IdentityProvider(Model):
+    """
+    An IdentityProvider is an instance of a provider.
+
+    For example, we may auto generate some, such as github.com, where it's
+    ``IdentityProvider(type='github', instance='github.com')``
+
+    When possible the instance should be the domain used.
+    """
+    __core__ = False
+
+    type = models.CharField(max_length=64)
+    instance = models.CharField(max_length=64)
+
+    class Meta:
+        app_label = 'sentry'
+        db_table = 'sentry_identityprovider'
+        unique_together = (('type', 'instance'),)
+
+
+class Identity(Model):
+    """
+    A unique identity with an external provider (e.g. GitHub).
+    """
+    __core__ = False
+
+    idp = FlexibleForeignKey('sentry.IdentityProvider')
```

Comment:
i think we'd also add a link to 'idp' in ``Integration``, though that may or may not be needed at a schema level since the instance is generated and is really just there to make foreign keys easy

its also possible we could not use the IdentityProvider model and simply put provider/instance on the Identity itself (though thats a bit gross)